src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,98 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+public class Storage {
+    private final String dukeData;
+
+    /**
+     * Constructor for Storage
+     *
+     * @param dukeData the String that contains the file path
+     */
+    public Storage(String dukeData) {
+        this.dukeData = dukeData;
+    }
+
+    /**
+     * Create the data file if it does not exist
+     *
+     * @return the list of tasks
+     * @throws IOException if the method txtToArrayList() fails
+     */
+    public ArrayList<Task> load() throws IOException {
+        File tempFile = new File("data/dukeData.txt");
+        if (tempFile.exists()) {
+            return txtToArrayList();
+        } else {
+            tempFile.createNewFile();
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * Converts a txt file of tasks to ArrayList of tasks
+     *
+     * @return an ArrayList of tasks
+     * @throws IOException if it is unable to read the file, file format is wrong
+     */
+    public ArrayList<Task> txtToArrayList() throws IOException {
+        ArrayList<Task> list = new ArrayList<>();
```

Comment:
Perhaps naming the Arraylist tasklist would be more descriptive?

Reply:
tasks may be appropriate as well